File: src/main/java/com/sem/keeper/service/LoanRequestService.java
package com.sem.keeper.service;

import com.sem.keeper.entity.DeviceEntity;
import com.sem.keeper.entity.LoanRequestEntity;
import com.sem.keeper.entity.UserEntity;
import com.sem.keeper.repo.LoanRequestRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;
import java.time.LocalDateTime;

@Service
@Transactional
public class LoanRequestService {

    @Autowired
    LoanRequestRepository loanRequestRepository;

    @Autowired
    LoanService loanService;

    public LoanRequestEntity addLoanRequest(UserEntity user, DeviceEntity deviceEntity){
        LoanRequestEntity neu = new LoanRequestEntity(user, deviceEntity);
        neu.setCreationDate(LocalDateTime.now());
        return loanRequestRepository.save(neu);
    }

    public void accept(LoanRequestEntity loanRequestEntity, UserEntity kiad) throws DeviceAlreadyOnLoanException {
        loanService.fromLoanRequest(loanRequestEntity, kiad);
        loanRequestRepository.delete(loanRequestEntity);
    }

    public void deny(long id){
        loanRequestRepository.deleteById(id);
    }
}

File: src/main/java/com/sem/keeper/service/LoanRequestService.kt
package com.sem.keeper.service

import com.sem.keeper.service.command.*
import com.sem.keeper.entity.LoanRequestEntity
import com.sem.keeper.repo.LoanRequestRepository
import org.springframework.stereotype.Service
import java.time.LocalDateTime
import javax.transaction.Transactional

@Service
open class LoanRequestService(
    open val loanRequestRepository: LoanRequestRepository,
    open val loanService: LoanService
) {
    @Transactional
    open fun addLoanRequest(command: NewLoanRequestCommand): LoanRequestEntity {
        val request = LoanRequestEntity(
            elvinne = command.user,
            deviceEntity = command.deviceEntity,
            creationDate = LocalDateTime.now()
        )
        return loanRequestRepository.save(request)
    }

    @Transactional
    open fun accept(command: AcceptLoanRequestCommand) {
        loanService.fromLoanRequest(command)
        loanRequestRepository.delete(command.loanRequestEntity)
    }

    @Transactional
    open fun deny(command: DenyLoanRequestCommand){
        when (command) {
            is DenyLoanRequestCommand.DenyLoanRequestByIdCommand -> loanRequestRepository.deleteById(command.id)
            is DenyLoanRequestCommand.DenyLoanRequestByEntityCommand -> loanRequestRepository.delete(command.loanRequestEntity)
        }
    }
}